roles/openshift_health_checker/library/etcdstats.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+'''
+Ansible module that determines if the size of paths in an etcd cluster exceed a given limit.
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+IMPORT_EXCEPTION = None
+
+try:
+    import etcd
+# pylint: disable=invalid-name
+except ImportError as IMPORT_EXCEPTION:
+    # Allows for test mocking
+    from collections import namedtuple
+    EtcdMock = namedtuple("etcd", ["EtcdKeyNotFound"])
+    etcd = EtcdMock(KeyError)
+
+
+# pylint: disable=too-many-arguments
+def check_etcd_key_size(client, key, total_size=0, size_limit=0, depth=0, depth_limit=1000, visited=None):
+    """Recursively checks that total size of an etcd path starting at a given key does not exceed a given limit"""
+    if visited is None:
+        visited = set()
+
+    if key in visited:
+        return 0, False
+
+    visited.add(key)
+
+    try:
+        result = client.read(key, recursive=False)
+    except etcd.EtcdKeyNotFound:
+        return 0, False
+
+    size = 0
+    limit_exceeded = False
+
+    for node in result.leaves:
+        if depth >= depth_limit:
+            raise Exception("Maximum recursive stack depth ({}) exceeded.".format(depth_limit))
+
+        if size_limit and total_size + size > size_limit:
+            return size, True
+
+        if not node.dir:
+            size += len(node.value)
+            continue
+
+        key_size, limit_exceeded = check_etcd_key_size(client, node.key,
+                                                       total_size + size,
+                                                       size_limit,
+                                                       depth + 1,
+                                                       depth_limit, visited)
+        size += key_size
+
+    return size, limit_exceeded or (total_size + size > size_limit)
```

Comment:
This is a confusing line.
Suggestion: give a name to the return value (create a variable), then `return size, name_of_bool_var`.